lib/PrintAsClang/PrintClangFunction.cpp

```diff
@@ -134,9 +134,9 @@ class ClangTypeHandler {
     if (auto *record = dyn_cast<clang::CXXRecordDecl>(typeDecl))
       return record->isTrivial();
 
-    // FIXME: If we can get plain clang::RecordDecls here, we need to figure out
-    //        how nontrivial (i.e. ARC) fields work.
-    assert(!isa<clang::RecordDecl>(typeDecl));
+    // Structs with ARC members are not considered trivial.
+    if (auto *record = dyn_cast<clang::RecordDecl>(typeDecl))
```

Comment:
Wouldn't a `clang::RecordDecl` always also be a `clang::CXXRecordDecl` when Clang is running in C++ language mode? That is, would this if-condition ever get triggered when there's another condition for `clang::CXXRecordDecl` above?

Reply:
That is correct! In `c++` mode we always get `CXXRecordDecl`s even for `C` structs. I believe the reason why we need to handel this case separately, because `in_add` in the test case is coming from `Foundation` and is defined in a header file that is processed in `C` mode. So, we end up consuming a C module that we reexpose C++ code.

Reply:
Ah, this is probably because the Foundation module was prebuilt in a C language mode. That's alright.